stock_reserve_area/models/stock_quant.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2023 ForgeFlow SL.
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+
+from odoo import api, fields, models
+from odoo.tools import float_compare
+
+
+class StockQuant(models.Model):
+    _inherit = "stock.quant"
+
+    reserve_area_id = fields.Many2one(
+        "stock.reserve.area", compute="_compute_reserve_area_id", store=True
+    )
+
+    @api.depends("location_id")
+    def _compute_reserve_area_id(self):
+        for quant in self:
+            reserve_areas = self.env["stock.reserve.area"].search([])
+            for reserve_area in reserve_areas:
+                if reserve_area.is_location_in_area(quant.location_id):
+                    quant.reserve_area_id = reserve_area
+                    break
+
+    def _get_reserve_area_available_quantity(self, product_id, reserve_area_id):
```

Comment:
Wouldn't this method make more sense in the stock.reserve.area model directly?

Reply:
I did it in here because it is like the _get_available_quantity form this model but specific for the reserve area.